Improved width validation in Sample Transmission Calculator

jclarkeSTFC opened the pull request:
**Description of work**
Add a check in the Sample Transmission Calculator for the bin width being too large so the user gets a nice message.

**Purpose of work**
<!-- Why has this work been done? If there is no linked issue please provide appropriate context for this work.
-->
Fixes #36196. 

<!-- If the original issue was raised by a user they should be named here. Do not leak email addresses
**Report to:** [user name]
-->

**Summary of work**
<!-- Please provide a short, high level description of the work that was done.
-->
- Added extra validation check to the Sample Transmission Calculator (the `single` option already had the same check, now it checks when specifying `multiple` as well).
- Improved error check on `single` case in Sample Transmission Calculator so it takes the start of the bin into account.

**To test:**

See #36196 for bug report. 

<!-- Instructions for testing.
There should be sufficient instructions for someone unfamiliar with the application to test - unless a specific
reviewer is requested.
If instructions for replicating the fault are contained in the linked issue then it is OK to refer back to these.
-->

<!-- alternative
*There is no associated issue.*
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

jclarkeSTFC committed:
Check bin width in Sample Transmission Calculator.

This check is already done for the single bin option input, but now it's
also done for the "multiple" input option. Width can't be greater than
max - min.

jclarkeSTFC committed:
Add more input validation to Rebin.

Check that width <= max - min when calling Rebin. Previously if that
happened then one bin was created centred at the middle of the interval.

jclarkeSTFC committed:
Add test for new width validation

Add two test cases to existing test that checks for bad Rebin input.

jclarkeSTFC committed:
Added release notes

jclarkeSTFC committed:
Add extra test case with more than three params.

jclarkeSTFC committed:
Take start of bin into account.

Error checks in the Sample Transmission Calculator need to take the
start of the bin into account as well.

jclarkeSTFC committed:
Remove extra width validation in RebinParamsValidator

This validation causes a problem with the Integration algorithm, because
Integration uses floating point min/max and deliberately tries to use a single
bin. The bin width must then be defined as exactly max-min, but the parameters
are passed to Rebin via a string, which causes a problem because max - min can be
very slightly more than max after loss of precision.

jclarkeSTFC committed:
Remove test added for Rebin validation.

jclarkeSTFC committed:
Adjusted release notes.

jclarkeSTFC committed:
No need for width check if max <= min already.

SilkeSchomann reviewed (changes requested):
It would be good to have a test for multiple wavelength range with width > max - min. At the moment, this is only tested for the single option.

jclarkeSTFC committed:
Added tests for multiple bin option.

Tests check that you get an error message if you try and specify multiple bin
input with a width larger than max - min.

jclarkeSTFC commented:
> It would be good to have a test for multiple wavelength range with width > max - min. At the moment, this is only tested for the single option.

Thanks, yes that's a good idea, I've added a couple of tests for the multiple option.
